index.js

```diff
@@ -614,8 +614,8 @@ function refFinder (ref, location) {
     const walk = ref[1].split('/')
     if (walk.length === 1) {
       const targetId = `#${ref[1]}`
-      let dereferenced = idFinder(schema, targetId)
-      if (dereferenced === undefined && !ref[0]) {
+      let dereferenced = idFinder(root, targetId)
```

Comment:
I understand this change, but I wonder if there is a more performant way to accomplish it such as:
it there is not the id in the `schema`, loop-up in the parent

Right now this change could impact performance and add new behaviours (same id in same schema)